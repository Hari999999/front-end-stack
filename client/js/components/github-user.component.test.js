import GithubUser from 'components/github-user.component';
import React from 'react';
import { mount, shallow } from 'enzyme';

function setup(testKey) {
    const props = {
        user: {
            getGithubUser: jest.fn(),
            user: {
                avatar: 'https://myimage.com', // eslint-disable-line camelcase
                username: 'hally9k',
                id: '123456'
            },
            error: null
        },
        error: {
            getGithubUser: jest.fn(),
            user: null,
            error: {
                message: 'ajax error 404'
            }
        }
    }[testKey];

    const enzymeWrapper = mount(<GithubUser {...props} />);

    return {
        props,
        enzymeWrapper
    };
}

describe('components', () => {
    describe('GithubUser', () => {
        it('should render the user when given a valid user object', () => {
            const { enzymeWrapper, props } = setup('user');
            expect(enzymeWrapper.find('label').text()).toBe('Github User: ');
            expect(enzymeWrapper.find('button').text()).toBe('Get User');
            expect(enzymeWrapper.find('div.github-user img').prop('src')).toBe('https://myimage.com');
            expect(enzymeWrapper.find('div.github-user h2').text()).toBe('hally9k');
            expect(enzymeWrapper.find('div.github-user h4').text()).toBe('ID: 123456');
            enzymeWrapper.find('button').simulate('click');
            expect(props.getGithubUser).toHaveBeenCalled();
        });
        it('should not render the user when given an error', () => {
            const { enzymeWrapper, props } = setup('error');
            expect(enzymeWrapper.find('label').text()).toBe('Github User: ');
            expect(enzymeWrapper.find('button').text()).toBe('Get User');
            expect(enzymeWrapper.find('div.github-user').exists()).toBe(false);
            expect(enzymeWrapper.find('div.error-block').exists()).toBe(true);
            enzymeWrapper.find('button').simulate('click');
            expect(props.getGithubUser).toHaveBeenCalled();
        });
        it('mounts the component', () => {
            const props = {
                getGithubUser: jest.fn(),
                user: {
                    avatar: 'https://myimage.com', // eslint-disable-line camelcase
                    username: 'hally9k',
                    id: '123456'
                },
                error: null
            };
            const enzymeWrapper = shallow(<GithubUser {...props} />);
            expect(enzymeWrapper).toMatchSnapshot();
        });
    });
});
