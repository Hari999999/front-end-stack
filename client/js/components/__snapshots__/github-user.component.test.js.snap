// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components GithubUser mounts the component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <label>
        Github User: 
        <input />
    </label>
    <button
        onClick={[Function]}
    >
        Get User
    </button>
    <div
        className="github-user"
    >
        <img
            src="https://myimage.com"
        />
        <h2>
            hally9k
        </h2>
        <h4>
            ID: 
            123456
        </h4>
    </div>
</div>,
  "nodes": Array [
    <div>
      <label>
            Github User: 
            <input />
      </label>
      <button
            onClick={[Function]}
      >
            Get User
      </button>
      <div
            className="github-user"
      >
            <img
                  src="https://myimage.com"
            />
            <h2>
                  hally9k
            </h2>
            <h4>
                  ID: 
                  123456
            </h4>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Root
        error={null}
        getGithubUser={[Function]}
        user={
                Object {
                        "avatar": "https://myimage.com",
                        "id": "123456",
                        "username": "hally9k",
                      }
        }
/>,
      "_debugID": 25,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Root {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "error": null,
          "getGithubUser": [Function],
          "user": Object {
            "avatar": "https://myimage.com",
            "id": "123456",
            "username": "hally9k",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 8,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <label>
                    Github User: 
                    <input />
          </label>
          <button
                    onClick={[Function]}
          >
                    Get User
          </button>
          <div
                    className="github-user"
          >
                    <img
                              src="https://myimage.com"
                    />
                    <h2>
                              hally9k
                    </h2>
                    <h4>
                              ID: 
                              123456
                    </h4>
          </div>
</div>,
        "_debugID": 26,
        "_renderedOutput": <div>
          <label>
                    Github User: 
                    <input />
          </label>
          <button
                    onClick={[Function]}
          >
                    Get User
          </button>
          <div
                    className="github-user"
          >
                    <img
                              src="https://myimage.com"
                    />
                    <h2>
                              hally9k
                    </h2>
                    <h4>
                              ID: 
                              123456
                    </h4>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Root
    error={null}
    getGithubUser={[Function]}
    user={
        Object {
            "avatar": "https://myimage.com",
            "id": "123456",
            "username": "hally9k",
          }
    }
/>,
}
`;
